src/modules/partition/gui/EncryptWidget.cpp

```diff
@@ -143,18 +146,24 @@ applyPixmap( QLabel* label, CalamaresUtils::ImageType pixmap )
 }
 
 void
-EncryptWidget::updateState()
+EncryptWidget::updateState( const bool notify )
 {
     if ( m_ui->m_passphraseLineEdit->isVisible() )
     {
         QString p1 = m_ui->m_passphraseLineEdit->text();
         QString p2 = m_ui->m_confirmLineEdit->text();
 
+
         if ( p1.isEmpty() && p2.isEmpty() )
         {
             applyPixmap( m_ui->m_iconLabel, CalamaresUtils::StatusWarning );
             m_ui->m_iconLabel->setToolTip( tr( "Please enter the same passphrase in both boxes." ) );
         }
+        else if ( FileSystem::typeForName( m_filesystem ) == FileSystem::Zfs && p1.length() < ZFS_MIN_LENGTH )
+        {
+            applyPixmap( m_ui->m_iconLabel, CalamaresUtils::StatusError );
+            m_ui->m_iconLabel->setToolTip( tr( "Password must be a minimum of %1 characters" ).arg( ZFS_MIN_LENGTH ) );
```

Comment:
is that a ZFS-specific requirement that always needs to be enforced? Otherwise, why not plug in existing password-checking mechanisms? (Er .. probably because they, klike libpwquality, are in the *user* module and not here).